### PR TITLE
fix Cookie.SameSite#parse to accept SameSite.None value

### PR DESCRIPTION
**Pull request by aparra**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Purpose

Fix Cookie.SameSite#parse to accept SameSite.None value

## References

https://web.dev/samesite-cookies-explained/

### CODE REVIEW

File: core/play/src/test/scala/play/api/mvc/CookiesSpec.scala [hunk 1]
```diff
@@ -337,4 +343,12 @@ class CookiesSpec extends Specification {
     }
   }
 
+  "object Cookie.SameSite#parse" should {
+    "successfully parse None value" in {
```
Comment (marcospereira) [inline core/play/src/test/scala/play/api/mvc/CookiesSpec.scala]: Add tests for the other values too. I don't think they exist now.

File: core/play/src/test/scala/play/api/mvc/CookiesSpec.scala [hunk 2]
```diff
@@ -156,6 +156,12 @@ class CookiesSpec extends Specification {
         Cookie("__Secure-ID", "123", secure = true, httpOnly = false, sameSite = Some(SameSite.Strict))
       )
     }
+    "handle SameSite=None cookie properly" in {
```
Comment (marcospereira) [inline core/play/src/test/scala/play/api/mvc/CookiesSpec.scala]: There is a similar test for `Strict` a few lines above. I would suggest to also add one for `Lax` (and have all the test names adjusted as well).

**Comment by aparra** [comment]
Hey @marcospereira. I've updated my PR to address your comments. Could you please review again? Thanks!

**Comment by notflorian** [comment]
Hello, would it be possible to backport this fix into a bugfix release 2.7.4?

Thanks :)

**Comment by kyle-wright** [comment]
Would be very beneficial to have this in 2.6 as well.

**Comment by renatocaval** [comment]
@decoded4620, the backport is happening. Travis is building on both branches (2.6.x and 2.7.x)

**Pull request merged**